python/paddle/trainer/config_parser.py

```diff
@@ -3850,6 +3859,16 @@ def __init__(self, name, inputs, reshape, **xargs):
             name, 'switch_order', 0, inputs=inputs, **xargs)
         self.config.reshape_conf.height_axis.extend(reshape['height'])
         self.config.reshape_conf.width_axis.extend(reshape['width'])
+        input_layer = self.get_input_layer(0)
+        if reshape is None:
+            self.set_layer_size(input_layer.size)
+        else:
+            inH = input_layer.height
+            inW = input_layer.width
+            inC = input_layer.size / inH / inW
+            out_dims = [0, inH, inW, inC]
+            size = reduce(lambda x, y: x * y, out_dims[reshape['width'][0]:])
```

Comment:
The input data for SwitchOrderLayer may be five dimensions (NCDHW).

Reply:
Thx.

Reply:
naming format: inH -> in_h, inW->in_w, ..

Reply:
Fixed.